server/cluster.go

```diff
@@ -1060,6 +1110,10 @@ func (sess *Session) clusterWriteLoop(forTopic string) {
 			// In both cases the msg does not need to be forwarded to the proxy.
 
 		case <-sess.detach:
+			log.Println("cluster: detach msg received", sess.sid)
+			return
+		default:
```

Comment:
Could you please check that this is not going to be called in a very tight loop.

Reply:
Fair enough. Here are some time deltas between successive calls for a session:
19.1µs
1.0314ms
283.1µs
5.8601ms
15.3µs
94.7µs
175µs
1.6739ms
1.7285ms
8.9µs

Let me look more into it.

Reply:
The `default` case in a `select` is run when no other case is ready. When channels are empty, the `for` will call `default` with no delay burning the CPU cycles.

Reply:
It should be either a ticker or no default.

Reply:
Well, yes. But there's a return statement on line 1118.
So the for loop stops once there are no more ready channels to read from and we exit the function.
What I'm more concerned about is when multiple `clusterWriteLoop` calls are scheduled in a rapid succession (hence, time deltas). In such a case, we'll have to deal with extra scheduling overhead.

Reply:
Why does it take so much time?
I've got a question on architecture.Originally we had a routine per topic-node. Say, 2K topics at 20 nodes gave 40K routines at each node, right? And supposedly it was a problem? 
Right now it's 20 routines shared between all nodes. What if we pin the go routines either to the number of topics or number of nodes * X?

Reply:
Sure. I'm planning to make the number of goroutines in the pool configurable.
I'll make it number of nodes * 5.